src/workerd/api/r2-bucket.c++

```diff
@@ -329,7 +329,7 @@ jsg::Promise<kj::Maybe<jsg::Ref<R2Bucket::HeadResult>>> R2Bucket::head(
 
     auto requestJson = json.encode(requestBuilder);
     kj::StringPtr components[2];
-    auto path = fillR2Path(components, adminAccount, adminBucket);
+    auto path = fillR2Path(components, nullptr, adminBucket);
```

Comment:
I think we should go ahead and remove this function or clean it up so it only expects the "bucket". 
Leaving it alone will probably lead to confusion.

Reply:
Otherwise the PR is looking pretty solid to me, thankfully this didn't seem too bad to fix.

Let's get an edgeworker PR up with the tests that show this working in action.